serde_derive/src/internals/attr.rs

```diff
@@ -129,25 +129,71 @@ impl<'c, T> VecAttr<'c, T> {
     }
 }
 
-pub struct Name {
-    serialize: String,
-    serialize_renamed: bool,
-    deserialize: String,
-    deserialize_renamed: bool,
-    deserialize_aliases: Vec<String>,
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Debug)]
+pub enum VariantName {
+    String(String),
+    Integer(i64),
+    Boolean(bool),
 }
 
-fn unraw(ident: &Ident) -> String {
-    ident.to_string().trim_start_matches("r#").to_owned()
+impl VariantName {
+    pub fn as_string(&self) -> Option<&String> {
+        match self {
+            VariantName::String(s) => Some(s),
+            _ => None,
+        }
+    }
+
+    pub fn as_int(&self) -> Option<i64> {
+        match self {
+            VariantName::Integer(i) => Some(*i),
+            _ => None,
+        }
+    }
+
+    pub fn as_bool(&self) -> Option<&bool> {
+        match self {
+            VariantName::Boolean(b) => Some(b),
+            _ => None,
+        }
+    }
+}
+
+impl ToString for VariantName {
```

Comment:
https://doc.rust-lang.org/1.71.0/std/string/trait.ToString.html: _"ToString shouldn’t be implemented directly"_